src/libsystemd/sd-event/sd-event.c

```diff
@@ -5279,7 +5271,6 @@ _public_ int sd_event_source_set_memory_pressure_type(sd_event_source *s, const
 
         free_and_replace(s->memory_pressure.write_buffer, w);
         s->memory_pressure.write_buffer_size = nl;
-        s->memory_pressure.locked = false;
```

Comment:
These encode an intention, and it's more robust against future refactors, so I'd rather keep them given it doesn't hurt